pkg/proxyserver/getter/logProxyGetter.go

```diff
@@ -0,0 +1,115 @@
+package getter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/stolostron/multicloud-operators-foundation/pkg/helpers"
+	"io"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"net/http"
+	clusterv1listers "open-cluster-management.io/api/client/cluster/listers/cluster/v1"
+)
+
+type LogProxyGetter struct {
+	ClusterLister      clusterv1listers.ManagedClusterLister
+	SecretLister       corev1listers.SecretLister
+	ProxyServiceHost   string
+	ProxyServiceCAFile string
+}
+
+func NewLogProxyGetter(clusterLister clusterv1listers.ManagedClusterLister,
+	secretLister corev1listers.SecretLister,
+	host, caFile string) *LogProxyGetter {
+	return &LogProxyGetter{
+		ClusterLister:      clusterLister,
+		SecretLister:       secretLister,
+		ProxyServiceHost:   host,
+		ProxyServiceCAFile: caFile,
+	}
+}
+
+func (c *LogProxyGetter) ProxyServiceAvailable(clusterName string) (bool, error) {
+	cluster, err := c.ClusterLister.Get(clusterName)
+	if err != nil {
+		return false, fmt.Errorf("failed to get cluster %s. %v", clusterName, err)
```

Comment:
notFound needs to be handled

Reply:
return false and no error if notFound

Reply:
use addonClient and return false and no error if notFound